http-client-spi/src/main/java/software/amazon/awssdk/http/DefaultSdkHttpFullRequest.java

```diff
@@ -349,6 +368,19 @@ private void copyHeadersIfNeeded() {
             }
         }
 
+        @Override
+        public SdkHttpFullRequest.Builder appendUriQueryParameter(String paramName, String paramValue) {
+            try {
+                this.uriQueryParameters.computeIfAbsent(paramName, k -> new ArrayList<>())
+                                       .add(URLDecoder.decode(paramValue, StandardCharsets.UTF_8.toString()));
+            } catch (UnsupportedEncodingException e) {
+                log.warn("Could not decode {}={} using {}", paramName, paramValue, StandardCharsets.UTF_8);
```

Comment:
We should just fail instead of logging a warning here

Reply:
method is no longer needed